### ISSUE TITLE
BUG: np.isnat fails on non-datetime

### ISSUE THREAD

**Issue by h-vetinari** [numpy/numpy#23582]
The function `np.isnat` asks a very specific question, which is what it says on the tin "is the argument the nan-marker for datetime/timedelta?"

There are situations -- especially in trying to write library code that survives a lot of user inputs (or tests for that) -- where it's necessary or at least desirable to handle NaT specifically. 

However the function unfortunately is way less useful than it could be, because it only works for datetime/timedelta:
```
TypeError: ufunc 'isnat' is only defined for np.datetime64 and np.timedelta64
```
so now, to use it somewhere (that's not in a hyperspecific chain of branches), I have to write
```
isinstance(x, (np.datetime64, np.timedelta64)) and np.isnat(x)
```
to achieve what `np.isnat(x)` could be doing in the first place.

### Why not make this function simply return False instead of TypeError in that case?

**Comment by seberg** [comment]
@jbrockmendel do you have a preference/intuition on this?

**Comment by jbrockmendel** [comment]
Either approach seems reasonable as long as it is consistent/documented.  On the margin I agree with @h-vetinari on the preferred behavior.  (I think i've seen similar unexpected raising coming from np.isnan)

**Comment by olinjohnson** [comment]
Hey - I'm interested in contributing to numpy. Can I work on a PR for this? Seems like a good first contribution.

**Comment by MatteoRaso** [comment]
>Can I work on a PR for this?

Certainly! Issues aren't assigned, so if you see something that needs to be fixed, you're always more than welcome to open a PR to help out. If you would like to know where to start when contributing to Numpy, there's a [great guide over at the website.](https://numpy.org/contribute/)

**Comment by caschb** [comment]
Hi, I stumbled upon this and decided to give it a shot

Looking for something similar I found gh-14841, which does the converse of what this change would do by having `datetime64`, `timedelta64` work with other functions such as `isnan` and `isinf`. 
That PR could serve as a guide for this change.

The question I have from that is, should the function still fail on types such as `str`, `bytes`, etc, or should it still still raise an exception like `isnan` does? @seberg 
I'd say yes, it should raise on those other types, but I don't know if you see it differently.

I was looking around for code within numpy that uses this and found that `assert_equeal` uses this function, however it would still it would work because now an `int` for example would return `np.False_`, so it seems that on that front there's no issue.
https://github.com/numpy/numpy/blob/ec6b2b2626d0fc2b1505f3ffd7905862ebe605f4/numpy/testing/_private/utils.py#L436-L450

However, this changes the API, and would have to rewrite tests that asserts a `TypeError` for this function, and change the documentation too.
